internal/injector/check.go

```diff
@@ -0,0 +1,34 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package injector
+
+import (
+	"fmt"
+	"go/ast"
+	"go/importer"
+	"go/token"
+	"go/types"
+	"runtime"
+)
+
+// typeCheck runs the Go type checker on the provided files, and returns the
+// Uses type information map that is built in the process.
+func (i *Injector) typeCheck(fset *token.FileSet, files []*ast.File) (map[*ast.Ident]types.Object, error) {
+	pkg := types.NewPackage(i.ImportPath, i.Name)
+	typeInfo := types.Info{Uses: make(map[*ast.Ident]types.Object)}
+
+	checkerCfg := types.Config{
+		GoVersion: i.GoVersion,
+		Importer:  importer.ForCompiler(fset, runtime.Compiler, i.Lookup),
```

Comment:
AFAIK - `importer.For` is deprecated in favor of `importer.ForCompiler`.

Reply:
And w/r/t `gcexportdata`, the part that is "faster" with `packages.Load` is the resolving; but we don't need the resolving part here because we have an extremely efficient `lookup` function (since we only look in the `importcfg` file to get to the archives).

Reply:
https://cs.opensource.google/go/go/+/refs/tags/go1.23.0:src/go/importer/importer.go;l=38

Reply:
Ah --- the deprecated behavior is specifically "if `lookup` is `nil`"